global_body_planner/include/global_body_planner/planning_utils.h

```diff
@@ -25,53 +26,134 @@
-  double F_MAX = 600;     // Maximum GRF, N (800 for cheetah, 500 for anymal)
-  double PEAK_GRF_MIN = 4.0;  // Minimum GRF in units of body weight
-  double PEAK_GRF_MAX = 7.0;  // Maximum GRF in units of body weight
-  double MU = 0.25;  // Friction coefficient (1.0 for Cheetah, 0.5 for ANYmal)
-  double T_S_MIN = 0.3;  // Minimum stance time, s
-  double T_S_MAX = 0.3;  // Maximum stance time, s
-  double T_F_MIN = 0.1;  // Minimum flight time, s
-  double T_F_MAX = 0.4;  // Maximum stance time, s
+  double M_CONST;       // Robot mass, kg
+  double J_CONST;       // Moment of inertia about the robot's y axis (pitch)
+  double G_CONST;       // Gravity constant, m/s^2
+  double F_MIN;         // Minimum GRF
+  double F_MAX;         // Maximum GRF, N
+  double PEAK_GRF_MIN;  // Minimum GRF in units of body weight
+  double PEAK_GRF_MAX;  // Maximum GRF in units of body weight
+  double MU;            // Friction coefficient
+  double T_S_MIN;       // Minimum stance time, s
+  double T_S_MAX;       // Maximum stance time, s
+  double T_F_MIN;       // Minimum flight time, s
+  double T_F_MAX;       // Maximum stance time, s
 
   // Define planning parameters
-  double KINEMATICS_RES =
-      0.03;  // Resolution of kinematic feasibility checks, m
-  double BACKUP_TIME =
-      0.3;  // Duration of backup after finding an invalid state, s
-  double BACKUP_RATIO =
-      0.5;  // Ratio of trajectory to back up after finding an invalid state, s
-  int NUM_GEN_STATES =
-      30;  // Number of actions computed for each extend function
-  double GOAL_BOUNDS = 0.5;  // Distance threshold on reaching the goal (only
-                             // used for vanilla RRT, not RRT-Connect)
-  double MAX_TIME = 2.0;     // Maximum planning time allowed
-  Eigen::Vector3d G_VEC;     // Maximum planning time allowed
+  double KINEMATICS_RES;      // Resolution of kinematic feasibility checks, m
+  int TRAPPED_BUFFER_FACTOR;  // Number of feasibility that must pass to not
+                              // consider a state trapped
+  double BACKUP_TIME;    // Duration of backup after finding an invalid state, s
+  double BACKUP_RATIO;   // Ratio of trajectory to back up after finding an
+                         // invalid state, s
+  int NUM_LEAP_SAMPLES;  // Number of actions computed for each extend function
+  double GOAL_BOUNDS;    // Distance threshold on reaching the goal (only
+                         // used for vanilla RRT, not RRT-Connect)
+  double MAX_TIME;       // Maximum planning time allowed
+  Eigen::Vector3d G_VEC;  // Maximum planning time allowed
+
+  // Define robot params and declare points used for validity checking
+  double ROBOT_L;  // Length of robot body, m
+  double ROBOT_W;  // Width of robot body, m
+  double ROBOT_H;  // Vertical distance between leg base and bottom of
+                   // robot, m
+
+  double body_traversability_threshold;
+  double contact_traversability_threshold;
```

Comment:
comment distinguishing these two would be great

Reply:
Done.